gnd/src/main/java/com/google/android/gnd/ui/editobservation/EditObservationFragment.java

```diff
@@ -289,6 +318,46 @@ private void onShowPhotoSelectorDialog(Field field) {
             }));
   }
 
+  private void showDateDialog(DateFieldViewModel fieldViewModel) {
```

Comment:
Can showing the date and time dialogs be unit tested using Roboelectric? (no need for end-to-end UI tests here, but a unit test would be helpful).

Reply:
added https://github.com/google/ground-android/pull/846/commits/f76f8c6727287e210a928fc4af2f71d8136205ce